app/models/g5_authenticatable/user.rb

```diff
@@ -38,7 +38,7 @@ def update_roles_from_auth(auth_data)
     end
 
     def clients
-      G5Updatable::ClientPolicy::Scope.new(self, G5Updatable::Client).resolve
+      G5Updatable::ClientPolicy::Scope.new(self, G5Updatable::Client).clients_from_client_and_location_roles
```

Comment:
Also.. having all this logic inside the `resolve` method makes it clear for other parts of the application to have the exact same behaviour elsewhere without the need to re-write all those instances using the new method.